### PR TITLE
fix: `ContentDialog` does not receive focus

### LINKED ISSUE #8980: [WPF] ContentDialog does not get Focus
### Current behavior

Currently on a WPF build a ContentDialog does not get focus/ prevent other elements from being interacted with. An example of this is having the ContentDialog pop up and then hitting tab to move through different elements. 

This does not happen in UWP.

[WPFDialogBoxNotFocus.zip](https://github.com/unoplatform/uno/files/8847556/WPFDialogBoxNotFocus.zip)
 

### Expected behavior

Expect for ContentDialog to take over page and prevent any element from being interacted with until 'Ok' or "Yes"/"No" is clicked. 

### How to reproduce it (as minimally and precisely as possible)

-Make Project
-Use ContentDialog
-When ContentDialog is open, hit tab
- You can now interact with page

### Workaround

N/A

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

WPF:
PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0"
PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0"
PackageReference Include="Uno.UI.Skia.Wpf" Version="4.2.0-dev.673"
PackageReference Include="Uno.UI.RemoteControl" Version="4.2.0-dev.673" Condition="'$(Configuration)'=='Debug'"
PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.2.0-dev.673"

WPF Host:
PackageReference Include="Uno.UI.Skia.Wpf" Version="4.2.0-dev.673"
PackageReference Include="Uno.UI.RemoteControl" Version="4.2.0-dev.673" Condition="'$(Configuration)'=='Debug'"
PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.2.0-dev.673"

UWP:
PackageReference Include="Microsoft.UI.Xaml" Version="2.6.2"
PackageReference Include="Uno.UI" Version="4.2.0-dev.673"
PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0"
PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="5.0.0"
PackageReference Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.0-dev.7"

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.2.0 Preview 3.0

### Relevant plugins

_No response_

### Anything else we need to know?

Have not tested other builds but did leave them in just in case.

### LINKED ISSUE #10135: ContentDialog focus is incorrect
Hi, I'm using a custom ContentDialog to get a string from a USB RFID reader. The reader acts as a keyboard and I use the KeyUp event handler to get the value:

`KeyUp += RfidDialog_KeyUp;` 

It works fine in WinUI but in the Skia project focus seems to be hooked in the background frame. You can see the white rectangle focusing the settings button.

[Focus rectangle](https://i.imgur.com/qshqdFO.png)

If I press Tab key, focus alternates between Settings and User button but the ContentDialog never receives the focus so it never gets KeyUp events.

Besides CornerRadius is displayed differently in both projects.

Here a link with both images:

[CornerRadius](https://imgur.com/a/ysKSgYD)

_Originally posted by @mcNets in https://github.com/unoplatform/uno/discussions/9545_

### LINKED ISSUE #10342: [Skia] Focused `TextBox` leaks
### Current behavior

When `TextBox` is focused at least once, it will never be reclaimed even after unloaded.

### Expected behavior

Should not leak.

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #10135, closes #8980, closes #10342

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

`ContentDialog` does not get focused

## What is the new behavior?

It does!

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs [hunk 1]
```diff
@@ -120,6 +124,61 @@ private void UpdateSizeProperties()
...
+							focusSet = m_tpSecondaryButtonPart.Focus(FocusState.Programmatic);
+						}
+						break;
+					case ContentDialogButton.Close:
+						if (m_tpCloseButtonPart is not null)
+						{
+							focusSet = m_tpCloseButtonPart.Focus(FocusState.Programmatic);
+						}
+						break;
+				}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs:169]:
```suggestion
				var button = DefaultButton switch
				{
					ContentDialogButton.Primary => m_tpPrimaryButtonPart,
					ContentDialogButton.Secondary => m_tpSecondaryButtonPart,
					ContentDialogButton.Close => m_tpCloseButtonPart,
					_ => null,
				}

				if (button is not null)
				{
					focusSet = button.Focus(FocusState.Programmatic);
				}
```

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs [hunk 2]
```diff
@@ -120,6 +124,61 @@ private void UpdateSizeProperties()
 			}
 		}
 
+		private void SetDialogFocus()
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs:127]: Are we able to runtime test it?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs:127]: Yes, on it 👍

**Comment by MartinZikmund** [comment]
Whoops 😂
image

**Pull request merged**